misc/swagger-ballerina/modules/ballerina-code-generator/src/main/java/org/ballerinalang/code/generator/CodeGenerator.java

```diff
@@ -0,0 +1,268 @@
+import org.ballerinalang.compiler.CompilerPhase;
+import org.ballerinalang.composer.service.ballerina.parser.service.model.BFile;
+import org.ballerinalang.composer.service.ballerina.parser.service.model.BallerinaFile;
+import org.ballerinalang.composer.service.ballerina.parser.service.model.lang.ModelPackage;
+import org.ballerinalang.composer.service.ballerina.parser.service.util.ParserUtils;
+import org.ballerinalang.model.tree.ServiceNode;
+import org.ballerinalang.model.tree.TopLevelNode;
+import org.wso2.ballerinalang.compiler.tree.BLangCompilationUnit;
+
+import org.wso2.ballerinalang.compiler.tree.BLangService;
+
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * <p>This class generates Service definitions, clients or connectors for a provided ballerina service.</p>
+ */
+public class CodeGenerator {
+
+    /**
+     * Generates code(client, server, connector etc) for ballerina source provided  in <code>definitionPath</code>
+     * <p>Method can be used for generating OAS 3.0.0, Swagger or Ballerina Client</p>
+     *
+     * @param type        Output type. Following types are supported
+     *                    <ul>
+     *                    <li>oas3</li>
+     *                    <li>swagger</li>
+     *                    <li>client</li>
+     *                    </ul>
+     * @param serviceNode Input Ballerina Service node to be process.
+     * @param outPath     Destination file path to save generated source files. If not provided
+     *                    <code>destinationPath</code> will be used as the default destination path'
+     * @throws CodeGeneratorException when file operations fail
+     * @Return generated source is string representation of generated source.
+     */
+    public String generateOutput(GeneratorConstants.GenType type, ServiceNode serviceNode, String outPath)
+            throws CodeGeneratorException {
+        ServiceNode context = serviceNode;
+        String output = "";
+        switch (type) {
+            case CLIENT:
+                output = getConvertedString(context, GeneratorConstants.DEFAULT_CLIENT_DIR,
+                        GeneratorConstants.CLIENT_TEMPLATE_NAME);
+                break;
+            case OPENAPI:
+                output = getConvertedString(context, GeneratorConstants.DEFAULT_OPEN_API_DIR,
+                        GeneratorConstants.OPEN_API_TEMPLATE_NAME);
+                break;
+            case SWAGGER:
+                output = getConvertedString(context, GeneratorConstants.DEFAULT_SWAGGER_DIR,
+                        GeneratorConstants.SWAGGER_TEMPLATE_NAME);
+                break;
+            default:
+                return output;
+        }
+        return output;
```

Comment:
Can we directly return from each case rather than breaking the switch and returning the value from here?

eg - `return getConvertedString(context, GeneratorConstants.DEFAULT_CLIENT_DIR, GeneratorConstants.CLIENT_TEMPLATE_NAME);`